src/imputation/MoR.py

```diff
@@ -0,0 +1,98 @@
+"""Functions for the Mean of Ratios (MoR) methods."""
+import pandas as pd
+import re
+
+from src.imputation.apportionment import run_apportionment
+
+good_statuses = ["Clear", "Clear - overridden"]
+bad_statuses = ["Form sent out", "Check needed"]
+
+
+def run_mor(df, backdata, target_vars):
+    """Function to implement Mean of Ratios method.
+
+    This is implemented by first carrying forward data from last year
+    for non-responders, and then calculating and applying growth rates
+    for each imputation class.
+
+    Args:
+        df (pd.DataFrame): Processed full responses DataFrame
+        backdata (pd.DataFrame): One period of backdata.
+        target_vars ([string]): List of variables to impute.
+
+    Returns:
+        pd.DataFrame: df with MoR applied
+    """
+    to_impute_df, remainder_df, backdata = mor_preprocessing(df, backdata)
+
+    # Carry forwards method
+    carried_forwards_df = carry_forwards(to_impute_df, backdata, target_vars)
+
+    # TODO Remove the `XXX_prev` columns (left in for QA)
+    return pd.concat([remainder_df, carried_forwards_df])
+
+
+def mor_preprocessing(df, backdata):
+    """Apply pre-processing ready for MoR
+
+    Args:
+        df (pd.DataFrame): full responses for the current year
+        backdata (pd.Dataframe): backdata file read in during staging.
+    """
+    # TODO move this to imputation main
+    # Select only values to be imputed and remove duplicate instances
+    imputation_condition = (
+        (df["formtype"] == "0001")
+        & (df["status"].isin(bad_statuses))
+        & (df["instance"] == 0 | pd.isnull(df["instance"]))
```

Comment:
We need to include ALL instances in the to_impute_df and then filter for instance 0 later. Otherwise we've got duplicates of instance 1 etc with CF in one of them and TMI in the other.